TASK: Add option to enable/disable creating event log entries for rec…

lorenzulrich opened the pull request:
…ords

Currently, if the Neos Event Log is active, besides general information about
the import run or the preset/part currently processed, the importer writes
three records to the event log for each item imported. If you have a lot of
records and/or the import is running frequently, this will make the eventlog
table growing quickly.

In case you have sufficient logging without the event log, you might want to
disable logging on a per-record basis. This change adds a new setting without
changing the default functionality. Refer to the README for more information.

lorenzulrich committed:
TASK: Add option to enable/disable creating event log entries for records

Currently, if the Neos Event Log is active, besides general information about
the import run or the preset/part currently processed, the importer writes
three records to the event log for each item imported. If you have a lot of
records and/or the import is running frequently, this will make the eventlog
table growing quickly.

In case you have sufficient logging without the event log, you might want to
disable logging on a per-record basis. This change adds a new setting without
changing the default functionality. Refer to the README for more information.

dfeyer commented:
Good one, I take a look more in details tomorrow

dfeyer commented:
@lorenzulrich After conflict resolution we can merge this one and release a new version, then works on the Neos 7.0 version

lorenzulrich commented:
I'm not sure anymore if this is a good contribution. After doing more digging, I found out that most records were not created because of the ContentRepositoryImporter itself, but because of the ContentRepository changes triggered by it (Update Node, Create Node etc.). Therefore I don't know if this makes sense.

For the customer affected by this, we now implemented another workaround for the event log: We added a switch (evaluated through AOP) to determine whether an action passed to the Eventlog was triggered by a user or not (which would mean by CLI etc.). In the latter case, we don't log:

```php
<?php

namespace My\FoobarCom\Aspects\Neos\EventLog;

/*
 * This file is part of the Neos.Neos package.
 *
 * (c) Contributors of the Neos Project - www.neos.io
 *
 * This package is Open Source Software. For the full copyright and license
 * information, please view the LICENSE file which was distributed with this
 * source code.
 */

use Neos\Flow\Annotations as Flow;
use Neos\Flow\Aop\JoinPointInterface;
use Neos\Neos\Domain\Model\User;
use Neos\Neos\Domain\Service\UserService;

/**
 * @Flow\Scope("singleton")
 * @Flow\Aspect
 */
class EventEmittingServiceAspect
{

    /**
     * @Flow\Inject
     * @var UserService
     */
    protected $userDomainService;

    /**
     * @var string
     */
    protected $currentUsername = null;

    /**
     * @Flow\InjectConfiguration(path="eventLog.enabled", package="Neos.Neos")
     * @var boolean
     */
    protected $enabled;

    /**
     * @Flow\InjectConfiguration(path="eventLog.onlyLogUserActions", package="Neos.Neos")
     * @var boolean
     */
    protected $onlyLogUserActions;

    /**
     * @Flow\Around("method(Neos\Neos\EventLog\Domain\Service\EventEmittingService->isEnabled())")
     * @return bool
     */
    public function isEnabled(JoinPointInterface $joinPoint)
    {
        $this->initializeCurrentUsername($joinPoint);

        if ($this->onlyLogUserActions && !$this->currentUsername) {
            return false;
        }

        return $this->enabled;
    }

    /**
     * Try to set the current username emitting the events, if possible
     *
     * @return void
     */
    protected function initializeCurrentUsername(JoinPointInterface $joinPoint)
    {
        if (isset($this->currentUsername)) {
            return;
        }

        $currentUser = $this->userDomainService->getCurrentUser();
        if (!$currentUser instanceof User) {
            return;
        }

        $this->currentUsername = $this->userDomainService->getUsername($currentUser);
    }
}
```

This change leads to more reduction of log entries.

What do you think?

dfeyer commented:
@lorenzulrich Fine, so a core contributation with a feature to disable the EventLog could be a nice idea, when done we can add support for this feature in the importer. Feel free to reopen if needed
